docs/standard/commandline/tab-completion.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Tab completion for System.CommandLine
+description: "How to enable and customize tab completion for apps built on the System.CommandLine library."
+ms.date: 02/22/2022
+no-loc: [System.CommandLine]
+helpviewer_keywords:
+  - "command line interface"
+  - "command line"
+  - "System.CommandLine"
+ms.topic: how-to
+---
+
+# Tab completion for System.CommandLine
+
+[!INCLUDE [scl-preview](../../../includes/scl-preview.md)]
+
+Apps that use `System.CommandLine` have built-in support for tab completion in certain shells. To enable it, the end user has to take a few steps once per shell. Tab completion is automatic for static values in your app, such as enum values or values you define by calling [FromAmong](define-commands.md#list-valid-argument-values). You can also customize the tab completion by getting values dynamically at runtime.
```

Comment:
@jonsequitur does tab-completion work dynamically for files? Should we include that here?

```suggestion
Apps that use `System.CommandLine` have built-in support for tab completion in certain shells. To enable it, the end user has to take a few steps once per shell. Once the user does this, tab completion is automatic for static values in your app, such as enum values or values you define by calling [FromAmong](define-commands.md#list-valid-argument-values). You can also customize the tab completion by getting values dynamically at runtime.
```